### DISCUSSION TITLE
Change release version naming from Stable to Release

### DISCUSSION THREAD

**Discussion by Flavelius** [godotengine/godot-proposals#7248] [General Discussions]
This might sound like a swing against godot (it's not), but it's interesting to note that other major game engines and other software don't typically use the term 'stable' in their release naming. It implies heavy QA and only minor outstanding issues. But most software, especially when open source or maintained by voluntary effort can and should probably not imply this by default.
A likely better fit would be to use 'Release'. This softens the expectation that quirks here and there may occur but that the reasonable assumption of a general usability with minor misbehaviours is given in this state where it's being released to the public.
It also conveniently follows the 'Release Candidate' nomenclature.

**Comment by AThousandShips** [comment]
I don't agree with the notion that "stable" implies anything of the sort, and "stable" indicates that no changes will be done to that specific version, which "release" does not

**Comment by Flavelius** [comment]
Wikipedia's definition mentions that software with this label typically has passed all stages of verification and tests and that remaining bugs are considered acceptable and that it's ready for production. In the case of such a big open project it's very subjective whether things are acceptable or production ready. But for the sake of progressing forward, in my opinion, every release is more a _confident snapshot_, even more with godot's accelerated release cycle now. LTS releases like 3.5, in my opinion warrant a 'stable' label.

**Reply by AThousandShips** [reply]
Well then that is the case? That's how release cycles work, known major blocking bugs are solved, that's how we operate

But also again its up for interpretation, but 4.1 is considered "ready for production", as is "4.0.3"

**Reply by Flavelius** [reply]
i've edited the last sentence. In my opinion 3.5 is an example for a good candidate for the stable label, because it carries the semantics of stability with the long term support label

**Reply by AThousandShips** [reply]
Well again, "stable" means that the version is released, and won't change

**Reply by Flavelius** [reply]
But betas and RC's typically also don't introduce new features. Godot did often break this - in a good way, but that indicates that labels are mostly a label, not a rule and in that case the associations with them gain more meaning than their static definitions. One of the issues i personally had/have with the stable label is that there were issues deep in basic asset handling that i often stumbled over that i would rate as breaking, so not production ready (some of which were scheduled to be fixed in a later release which is good, but in my opinion a discussion worthy handling of the stable label). But this is a matter of personal opinion of course how to rate certain breaking bugs or known issues that don't make a full stable release in the 'stability' sense but for acceptability or broad usage. That's why i thought it to be a better fit for a discussion than a proposal.

**Reply by AThousandShips** [reply]
So, if this is a question of changing "stable" for "release" I say that's just semantics and unhelpful

If it is changing to using both "stable" and "release" then I see that as a major hurdle, how do we decide what gets what label? What is the impact of that? Will people hesitate to use versions labelled "release"? What metric do we use? How much negative feedback will "promising" a "stable" version produce if some major bug pops up after release?

I don't see it being worthwhile, I feel that it either will lead to never releasing "stable" versions, or the label being meaningless, bugs will always exist, and the ability to solve them is entirely based on people being able to identify them and being willing to write the code to solve them. We have no ability to ensure that specific bugs are solved (not that any development environment can, or should, promise that) but being a community driven project we are at the mercy of these parameters.

We can't "plan" for solving bugs, and therefore I feel that "stable" is impossible, and misleading as it only says "we have cleared out all the glaring bugs", which is just a slightly different classification of what we already use for releases, we aim to eliminate bugs that are considered blocking, and will hold off releasing if something pops up

**Reply by Flavelius** [reply]
As i said, 3.5 ('**LTS**') is a good candidate, 3.5 with it's feature set will stay 3.5, it mainly gets fixes, so it can be considered stable. 4.x is a constantly changing target. it's more like a rolling release. feature wise now in 4.x betas are not that different from stables apart from a few more acknowledged open tasks.
In the same way 4.0 could be LTS stable.
Similar to unity where they have tech stream releases (inbetween versions) and LTS ones, where LTS versions are considered stable and ready for production and tech streams are not.

**Reply by AThousandShips** [reply]
Well 4.x isn't stable, 4.1.0 is, and 4.0.3 is, those never change, you're mixing LTS with stable

> We call major.minor combinations stable branches. Each stable branch starts with a major.minor release (without the 0 for patch) and is further developed for maintenance releases in a Git branch of the same name (for example patch updates for the 4.0 stable branch are developed in the 4.0 Git branch).

As you say, it's a matter of preference, and I think the issues I've expressed are relevant, and I don't see the value in this sort of change, it would be relatively irrelevant, hard to define, and wouldn't provide any stronger guarantee than we already do

**Reply by Flavelius** [reply]
What i meant was that if 4.0 was an LTS release it could be considered stable. But the way it currently presents itself is that it's also more like unity's tech streams, where each 4.x is just another sub-release, where none are actually fixed, because if all are fixed, there's no difference to betas. Meaning beta1 also didn't change after beta2 came out, as beta2 was worked on at that time.

**Reply by AThousandShips** [reply]
So you're using "stable" just to mean LTS, but we already call them LTS so what's the difference?

**Reply by Flavelius** [reply]
None. I didn't want to propose changing them, LTS is fine. But stable has a different meaning to a constantly changing target, that is 4.x in this case.

**Reply by AThousandShips** [reply]
But 4.x isn't claimed to be stable? But 4.1.0 is, it's equally stable to 3.5.2

**Reply by Flavelius** [reply]
Then seeing all the 'stable' occurrences in the name of each minor 4.x sub-release is definitely confusing https://downloads.tuxfamily.org/godotengine/ if they don't claim the same stablility as a full 4.1 or 3.5 / LTS release.
It basically comes down to the semantics of stable being stretched a bit much or it being applied too liberally (this way beta1 could be considered stable as well, as it doesn't change anymore after beta2 is being worked on)

**Reply by AThousandShips** [reply]
4.1.0 has the same stability as 3.5.2 by the definition given, not the assumption of any given individual 

Stable as opposed to under development, as opposed to pre-release

**Comment by dalexeev** [comment]
This change could break CI and similar tools of third party projects, distribution repositories, etc., so we shouldn't change this unless absolutely necessary. This might be acceptable at the very beginning of the major release cycle, but it's too late to change it now, in my opinion.

"stable" in this context is a synonym for "release". The term "release" is used more frequently, such as the name of the category on the blog. "stable" is more of a technical term that can be confusing to new users, indeed. But I don't think this is a problem, as new users usually start selecting the Godot version on the website's download page and don't see the technical term at all. If we need to improve something to help users choose the Godot version, then we need to look there.

Also note that "stable" does not mean "more stable" or "less stable" relative to other **releases**. This means "stable enough" compared to dev, alpha, beta, and release candidates. We consider the 4.x releases to be "stable enough" to be used in many cases. Even if 3.x releases are "more stable". This is emphasized by the "LTS" status, not "stable".

**Reply by Flavelius** [reply]
That's a good explanation. Maybe my issue finds a good description in the way that official communication for godot is not specifically designed (as opposed to the marketing of other engines) but just flows naturally between technical and non-technical in the godot community by the nature of being somewhat connected to the internal engine development.

**Reply by Calinou** [reply]
`stable` can also be understood in the context of "not changing" rather than a lack of bugs, similar to Debian's terminology.

**Reply by Flavelius** [reply]
That's where a part of my issue comes from, as godot is constantly changing (apart from the LTS). If not-changing applies to 4.0.2, 4.0.3 etc. then the semantics of not-changing as stable become meaningless, as any beta[X] is also not changing anymore after beta[X+1] is being worked on. But it does make sense as a technical label for internal management as the op here said.

**Reply by AThousandShips** [reply]
"4.1-beta1" doesn't exist in the codebase though, there's no "4.1-beta1" tag, it only exists as a blog post and similar, the repository only tags things as "4.1-dev", "4.1-beta", "4.1-rc", etc.

**Reply by Flavelius** [reply]
Maybe my wording was bad, it doesn't change the meaning i wanted to convey. I've adjusted the text.
